File: test2/Question8.py
x = 1
y = 3
z = 5

# nested first deep
def sub1():
    a = 7
    y = 9
    z = 11
    print("First", a,y,z)
    # nested second deep
    def sub2():
        global x
        a= 13
        x = 15
        w = 17
 
        print("global", a,x,w)

        # nested third deep
        def sub3():
            nonlocal a
            a = 19
            b = 21
            z = 23
            print("local", a,b,z)
        sub3()
    sub2()
sub1()

File: test2/question8.js
var x = 1, 
    y = 3,
    z = 5;
    sub1();
function sub1(){
    var a = 7, 
        y = 9,
        z = 11;
    console.log("first " , a, y, z + "");
    sub2()
}
    function sub2(){
        var a = 13,
            b = 15,
            z = 17;
        console.log("global " , a, b, z + "");
        return sub3()
    }
function sub3(){
    var a = 19,
        x = 21,
        w = 23;
        console.log("local " , a, x, w, "");
}